src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -49,6 +52,75 @@ public static Encoding OutputEncoding
             get { return GetConsoleEncoding(); }
         }
 
+        private static readonly object InternalSyncObject = new object();
```

Comment:
Nit: InternalSyncObject => s_internalSyncObject
I'm also not clear on the "internal" part of the name and why that's necessary.